infra/modules/kubevela-addons/definitions/cardano-wallet-connector.cue

```diff
@@ -0,0 +1,171 @@
+parameter: {
+	network:       *"preview" | "preprod" | "mainnet"
+}
+
+#cardanoWalletConfigs: {
+  volumes: [{
+    name: "wallet-db"
+    persistentVolumeClaim: {
+      claimName: "wallet-db"
+    }
+  }, {
+    name: "db"
+    persistentVolumeClaim: {
+      claimName: "db"
+    }
+  }]
+  volumeMounts: [{
+    name:      "wallet-db"
+    mountPath: "/wallet-db"
+    storageClassName: "ebs-sc"
+    resources: {
+      requests: {
+        storage: "100Ki"
+      }
+    }
+  }, {
+    name:      "db"
+    mountPath: "/db"
+    storageClassName: "ebs-sc"
+    resources: {
+      requests: {
+        storage: "100Ki"
+      }
+    }
+  }]
+}
+
+#cardanoNodeConfigs: {
+  volumes: [{
+    name: "ipc"
+    emptyDir: {}
+  }]
+  envs: [{
+    name: "CARDANO_NODE_SOCKET_PATH"
+    value: "/ipc/node.socket"
+  }]
+}
+
+#PatchConfig: {
+  volumes: [#cardanoNodeConfigs.volumes + #cardanoWalletConfigs.volumes][0]
+  envs: [#cardanoNodeConfigs.envs][0]
+}
+
+patch: spec: template: spec: {
+	// +patchKey=name
+	volumes: #PatchConfig.volumes
+  // +patchKey=name
+  volumeMounts: #cardanoWalletConfigs.volumeMounts,
+	// +patchKey=name
+	containers: [
+		{
+			name: context.name
+			volumeMounts: [{
+				name:      #cardanoNodeConfigs.volumes[0].name
+				mountPath: "/\( #cardanoNodeConfigs.volumes[0].name )"
+			}]
+	    // +patchKey=name
+			env: #PatchConfig.envs
+		},
+		{
+			name:            "socat"
+			image:           "alpine/socat"
+			imagePullPolicy: "Always"
+			args: [
+				"UNIX-LISTEN:\( #cardanoNodeConfigs.envs[0].value ),fork",
+				"TCP-CONNECT:cardano-node-\( parameter.network ).vela-system:8090",
+			]
+			volumeMounts: [{
+				name:      #cardanoNodeConfigs.volumes[0].name
+				mountPath: "/\( #cardanoNodeConfigs.volumes[0].name )"
+			}]
+		},
+    {
+      name:            "cardano-wallet-\( parameter.network )"
+      image:           "inputoutput/cardano-wallet:dev-master"
+      imagePullPolicy: "Always"
+      args: [
+        "serve",
+        "--node-socket",
+        "/ipc/node.socket",
+        "--database",
+        "/wallet-db",
+        "--listen-address",
+        "0.0.0.0",
+        "--testnet",
+        "/config/\( parameter.network )/genesis-byron.json"
+      ]
+      volumes: [{
+        name:      #cardanoNodeConfigs.volumes[0].name
+        path: "/\( #cardanoNodeConfigs.volumes[0].name )"
+      }, {
+        name:      #cardanoWalletConfigs.volumes[0].name
+        path: "/\( #cardanoWalletConfigs.volumes[0].name )"
+      }]
+      volumeMounts: [{
+        name:      #cardanoNodeConfigs.volumes[0].name
+        mountPath: "/\( #cardanoNodeConfigs.volumes[0].name )"
+        emptyDir: {}
+      }]
+    },
+    {
+      name:            "create-wallet"
```

Comment:
Why is this here? Leftover from testing?

Reply:
ops, yes. thank you!

Reply:
It would be nice to have some mechanism to create a wallet automatically, ideally it shouldn't be a sidecar container because only the first execution will be succeed, the other ones will fail and this container will always be with `CrashLoopBackOff` state.

```
% kubectl --context scde-dev -n dapps-certification-staging logs dapps-certification-6dd5b68499-q5plq -c create-wallet; echo
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   472    0   194  100   278   8864  12703 --:--:-- --:--:-- --:--:-- 22476
{"code":"wallet_already_exists","message":"This operation would yield a wallet with the following id: 73857344a0cf884fe044abfe85660cc9a81f6366 However, I already know of a wallet with this id."}
```

Reply:
There shouldn't be any wallet creation here at all, any more than `sqitch` calls should be part of the RDS component

Reply:
I am not sure if I can agree with that because without it we have to create that wallet manually when we are deploying this application for the first time. We don't have to do nothing manually with the RDS component, it just works. :P 

Is it ok to have such manual step? If yes, it's ok for me as well.

Reply:
With the RDS component, you have to initialize your DB schema, right? In any case, yes I think it's OK, certainly it's not OK to require a wallet seed phrase  in your config :grimacing:

Reply:
I don't have to. The RDS is creating an empty database and all the schemas are initialized by the applications itself. Probably we will have problems with applications which don't have that kind of schema migration/initialization tho. 😢 

Ok, I agree with you! Removing my request for change. :)